### PR TITLE
Use node16 in ci

### PR DESCRIPTION
**Pull request by andre4i**
## Description

Resubmitting https://github.com/microsoft/FluidFramework/pull/16271 after it was reverted, with two additional changes:

- use ` $(Build.SourcesDirectory)` as the working directory
- extract the node version in a single variable

### CODE REVIEW

File: tools/pipelines/templates/include-vars-node-version.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+# Licensed under the MIT License.
+
+variables:
+# The node.js version to be used
+- name: nodeVersion
+  value: '16.16.0'
```
Comment (andre4i) [inline tools/pipelines/templates/include-vars-node-version.yml:7]: @alexvy86 alternatively we could use a default parameter for the `tools/pipelines/templates/include-use-node-version.yml` and not provide it at all in the tasks but this seemed more explicit. Let me know what you think.

Comment (alexvy86) [reply tools/pipelines/templates/include-vars-node-version.yml:7]: I think I lean towards having it be the default in the `use` template. It gives us the same flexibility and we don't need to import the `vars` file in every pipeline. I personally dislike variable-definition templates a bit, because they introduce variables in a way that is not obvious at all for someone reading through the pipeline definition (and similar complaint with variable groups).

If we wanted to test _everything_ with a new version of node we'd have to make a change either way, in the `var` template or in the `use` template. If we wanted to test it for a particular pipeline, we could still provide the override in it in both cases.

I'd see the point of the vars file if the variable was used for different purposes in different pipelines, but if everywhere we use it is only to pass it to the use-node task... I think it's better to just put it there directly.

File: tools/pipelines/templates/include-use-node-version.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+# Licensed under the MIT License.
+
+# include-use-node-version template to use a specific node version
+
+parameters:
+# The node version required. The version must already be installed in the image.
```
Comment (alexvy86) [inline tools/pipelines/templates/include-use-node-version.yml:7]: Hmm does it really? Could we make this task install it if not found? Baking the supported ones into the image still makes sense so they don't have to be downloaded, but installing a not-found one simplifies the scenario @tylerbutler brought up of testing against a new version that might not be already in the image (and would also go back to paying the price of potential flakiness, but that seems acceptable for testing).

File: tools/pipelines/templates/include-vars-node-version.yml
Comment (tylerbutler) [inline tools/pipelines/templates/include-vars-node-version.yml:1]: I can't think of a good reason not to put the contents of this file into `include-vars.yml` instead. Is there a reason it needs to be separate?

File: tools/pipelines/templates/include-install-pnpm.yml [hunk 3]
```diff
@@ -44,7 +44,8 @@ steps:
     condition: or(succeeded(), canceled())
     # workspace-concurrency 0 means use use the CPU core count. This is better than the default (4) for larger agents.
     script: |
-      corepack enable
-      pnpm -v
+      echo "Using node $(node --version)"
+      sudo corepack enable
```
Comment (tylerbutler) [inline tools/pipelines/templates/include-install-pnpm.yml:48]: Why is this needed?

Comment (andre4i) [reply tools/pipelines/templates/include-install-pnpm.yml:48]: @tylerbutler because of https://dev.azure.com/fluidframework/public/_build/results?buildId=174472&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=996a989a-edd9-5084-d9de-d81350de1bfc

File: tools/pipelines/templates/include-use-node-version.yml (hunk 2 above)
Comment (tylerbutler) [reply tools/pipelines/templates/include-use-node-version.yml:7]: I'm OK with whatever for expediency, but I would like to reach a conclusion on the Teams thread so we can establish our relative priorities for things like this. If that leads us to want to change this (as I do) then we can follow up with whatever change is needed. The only drawback to that I can think of is that some release branches will be stuck on "old CI config" for a bit if we change it again.

Comment (andre4i) [reply tools/pipelines/templates/include-use-node-version.yml:7]: How do we enforce the testing part? I could totally see us going back to installing it all the time as opposed to building the image again. For testing, someone can alter this script and install it explicitly.

**Review by alexvy86 — APPROVED** [review]
